File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    dct = {}
    answer = ''
    for i in participant:
        try:
            dct[i] += 1
        except:
            dct[i] = 1

    for j in completion:
        dct[j] -= 1
        if dct[j] == 0:
            del dct[j]
    
    ans = list(dct.keys())
    answer = ans[-1]
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> partMap = new HashMap<>();
        
        for (String p : participant) {
            partMap.put(p, partMap.getOrDefault(p, 0) + 1);
        }
        
        for (String c : completion) {
            partMap.put(c, partMap.get(c) - 1);
            if (partMap.get(c) == 0) {
                partMap.remove(c);
            }
        }

        answer = partMap.keySet().iterator().next();
        
        return answer;
    }
}